### PR TITLE
Add read-ai extension

### PR DESCRIPTION
**Pull request by JungHoonGhae**
## Description

Feeling too lazy to read? No worries, we've got you covered! This Raycast extension is your go-to solution for those moments when reading feels like a chore. Leveraging OpenAI's Text-to-Speech (TTS) API, it transforms selected text into speech in a style of your choice. Perfect for those who prefer to listen, as well as for visually impaired users who benefit from auditory content. Sit back, relax, and let the extension do the reading for you, in the way you like it.

### Features

**1. Advanced Text-to-Speech**
Dynamic Reading Styles: Go beyond traditional text-to-speech with a range of reading styles. Choose to have your text read in a:
- Auto Detection Style: Automatically detects the most suitable reading style based on the content.
- Summary Style: Provides a concise summary of the text.
- Detailed Style: Offers a detailed explanation of the text.
- Audio Book Style: Delivers a rich and immersive storytelling experience.
- Educational Style: Explains the text in an educational and informative manner.
- Explain Code Style: Ideal for developers, this style focuses on clearly explaining code snippets.
- Code Review Style: Get your code reviewed out loud with constructive feedback and commentary.
- News Style: For an informative and formal tone, similar to a newscaster.
- Podcast Style: Enjoy a conversational and engaging tone, akin to listening to a podcast.
- Tweet Style: Perfect for reading out social media posts or short messages.
- SparkNotes Style: Provides a quick and easy understanding of complex literature.
- And more!

**2. Adapt to Content Typ**e: The extension intelligently adapts its reading style based on the type of content, enhancing comprehension and engagement.

**3. Personalization and Accessibility**
Custom Voice Selection: Choose from a variety of voices to find the one that best suits your listening preference, whether it's for casual listening or professional use.

**4. Language Translation**: Broaden your horizons by translating and listening to content in different languages.

**5. Subtitles Option**: For those who prefer visual aid, enable subtitles to read along with the spoken text, improving accessibility and understanding.

**6. Customizable AI Settings**: Tailor the text-to-speech output to your liking by adjusting AI response temperature, selecting the preferred GPT model, and more.

### Enhanced User Experience

1. **Ideal for Various User Groups**: Whether you're a student, a professional, a developer, or someone who enjoys listening to content rather than reading it, this extension is designed to cater to a wide array of needs.

2. **Accessibility for Visually Impaired**: This extension is particularly useful for visually impaired users, offering them a means to access written content in an audible format.

## Screencast

https://github.com/raycast/extensions/assets/42439321/b1b746e0-cdd1-4409-84b4-3086df8ed284

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by JungHoonGhae** [comment]
I have made all the necessary changes and committed them. If there are any issues after the review, please let me know so that I can make the necessary corrections.

**Comment by JungHoonGhae** [comment]
I have made all the necessary changes and committed them. If there are any issues after the review, please let me know so that I can make the necessary corrections.

**Comment by JungHoonGhae** [comment]
I have made all the necessary changes and committed them. If there are any issues after the review, please let me know so that I can make the necessary corrections.

**Comment by JungHoonGhae** [comment]
I have made all the necessary changes and committed them. If there are any issues after the review, please let me know so that I can make the necessary corrections. 💪🏼

**Comment by JungHoonGhae** [comment]
I have made all the necessary changes and committed them. If there are any issues after the review, please let me know so that I can make the necessary corrections. 💪🏼

**Comment by JungHoonGhae** [comment]
I have made all the necessary changes and committed them. If there are any issues after the review, please let me know so that I can make the necessary corrections. 💪🏼

**Comment by JungHoonGhae** [comment]
I have made all the necessary changes and committed them. If there are any issues after the review, please let me know so that I can make the necessary corrections. 💪🏼

**Comment by JungHoonGhae** [comment]
@pernielsentikaer I have made all the required edits and committed them. If any issues arise during the review, please inform me so that I can promptly rectify them.  💥

**Comment by JungHoonGhae** [comment]
I have made all the required edits and committed them. If any issues arise during the review, please inform me so that I can promptly rectify them. 💥

**Comment by DophinL** [comment]
WOW, that's really interesting.Does it support Chinese? @JungHoonGhae

**Comment by JungHoonGhae** [comment]
> WOW, that's really interesting.Does it support Chinese? @JungHoonGhae

Sure! Since I am just using OpenAI's TTS (Text-to-Speech) model it supports a wide range of languages, including: Afrikaans, Arabic, Armenian, Azerbaijani, Belarusian, Bosnian, Bulgarian, Catalan, Chinese, Croatian, Czech, Danish, Dutch, English, Estonian, Finnish, French, Galician, German, Greek, Hebrew, Hindi, Hungarian, Icelandic, Indonesian, Italian, Japanese, Kannada, Kazakh, Korean, Latvian, Lithuanian, Macedonian, Malay, Marathi, Maori, Nepali, Norwegian, Persian, Polish, Portuguese, Romanian, Russian, Serbian, Slovak, Slovenian, Spanish, Swahili, Swedish, Tagalog, Tamil, Thai, Turkish, Ukrainian, Urdu, Vietnamese, and Welsh. @DophinL

File: extensions/read-ai/src/types/index.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+export interface Preferences {
+  apiKey: string;
+  defaultVoice: Voice;
+  temperature: string;
+  gptModel: string;
+  subtitlesToggle: boolean;
+  outputLanguage: LanguageCode;
+  readingStyle: readingStyle;
+}
```
Comment (pernielsentikaer) [inline extensions/read-ai/src/types/index.ts:14]: This is already built-in inside the folder in `raycast-env.d.ts`

File: extensions/read-ai/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,277 @@
+import { getSelectedText, showToast, Toast, getPreferenceValues, closeMainWindow } from "@raycast/api";
+import { OpenAI } from "openai";
+import fs from "fs/promises";
+import path from "path";
+import os from "os";
+import { LanguageCode, readingStyle, Voice, Preferences } from "./types";
```
Comment (pernielsentikaer) [inline extensions/read-ai/src/index.ts:6]:
```suggestion
import { LanguageCode, readingStyle, Voice } from "./types";
```

File: extensions/read-ai/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+      this.isPlaying = true;
+      const { filePath, text } = this.playAudioQueue.shift()!;
+      try {
+        console.log("Playing audio:", filePath, text);
+        if (this.subtitlesToggle) {
+          showToast({
+            style: Toast.Style.Animated,
+            title: `💬 ${text}`,
+            message: text,
+          });
```
Comment (pernielsentikaer) [inline extensions/read-ai/src/index.ts:207]:
```suggestion
          showToast({
            style: Toast.Style.Animated,
            title: `💬 ${text}`,
          });
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you 🙂

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by JungHoonGhae** [comment]
I've carefully addressed all the points in your feedback and made the necessary changes. 🛠️ Looking forward to your review of the updated extension. Thank you for your guidance! 👍🙂 @pernielsentikaer

**Review by JungHoonGhae — COMMENTED** [review]
I've carefully addressed all the points in your feedback and made the necessary changes. 🛠️ Looking forward to your review of the updated extension. Thank you for your guidance! 👍🙂 @pernielsentikaer

**Review by JungHoonGhae — COMMENTED** [review]
I've carefully addressed all the points in your feedback and made the necessary changes. 🛠️ Looking forward to your review of the updated extension. Thank you for your guidance! 👍🙂 @pernielsentikaer

**Review by pernielsentikaer — APPROVED** [review]
✅ Approved

**Pull request merged**